src/main/java/net/minecraftforge/fml/common/registry/VillagerRegistry.java

```diff
@@ -285,9 +324,46 @@ public VillagerCareer(VillagerProfession parent, String name)
 
         private VillagerCareer init(EntityVillager.ITradeList[][] traids)
         {
+            for(int i = 0; i<traids.length; i++)
+            {
+                ArrayList<ITradeList> tradesForLevel = new ArrayList<ITradeList>();
+                for(ITradeList trade: traids[i])
+                {
```

Comment:
Protip, arraylist has a constructor that takes and array, also all lists have a addAll function.